### ISSUE TITLE
what is the final avg loss value?

### ISSUE THREAD

**Issue by dedoogong** [ppengtang/oicr#12]
I'm trying to modify your code and retrain it with VOC 2007 first and VOC2012 and COCO later. 
Currently, I'm using 2 GPU(each 8GB, so I changed the 5 multiscale factors to 4 (480, 576, 688, 864) and MAX_SIZE to 1024) and from around 24000 iterations, the loss values were dropped to around 0.1xxx to 0.2xxx( it starts from around 4.xx to 5.xx ).
I wonder this result is correct or not. 

Can anyone share the training loss results?

Thank you~!

10:43.547068  Iteration 4260, lr = 0.0001
11:03.822432 32388 solver.cpp:219] Iteration 4280 (0.986387 iter/s, 20.276s/20 iters), loss = 0.171675
11:03.822456      Train net output #0: loss_midn = 0.243829 (* 1 = 0.243829 loss)
11:03.822463      Train net output #1: loss_refine = 2.31813e-19 (* 1 = 2.31813e-19 loss)
11:03.822466      Train net output #2: loss_refine1 = 0.00681272 (* 1 = 0.00681272 loss)
11:03.822470      Train net output #3: loss_refine2 = 0.00233187 (* 1 = 0.00233187 loss)
11:03.822859  Iteration 4280, lr = 0.0001
11:22.718451 32388 solver.cpp:219] Iteration 4300 (1.05841 iter/s, 18.8962s/20 iters), loss = 0.17244
11:22.718477      Train net output #0: loss_midn = 0.0021711 (* 1 = 0.0021711 loss)
11:22.718483      Train net output #1: loss_refine = 0.00475247 (* 1 = 0.00475247 loss)
11:22.718487      Train net output #2: loss_refine1 = 0.0212138 (* 1 = 0.0212138 loss)
11:22.718490      Train net output #3: loss_refine2 = 0.0162532 (* 1 = 0.0162532 loss)
11:22.829041  Iteration 4300, lr = 0.0001
11:41.827881 32388 solver.cpp:219] Iteration 4320 (1.04659 iter/s, 19.1096s/20 iters), loss = 0.196953
11:41.827906      Train net output #0: loss_midn = 0.0358219 (* 1 = 0.0358219 loss)
11:41.827913      Train net output #1: loss_refine = 2.13161e-16 (* 1 = 2.13161e-16 loss)
11:41.827916      Train net output #2: loss_refine1 = 0.0165497 (* 1 = 0.0165497 loss)
11:41.827919      Train net output #3: loss_refine2 = 0.0209345 (* 1 = 0.0209345 loss)
11:41.828313  Iteration 4320, lr = 0.0001
12:01.396237 32388 solver.cpp:219] Iteration 4340 (1.02205 iter/s, 19.5686s/20 iters), loss = 0.215037
12:01.396260      Train net output #0: loss_midn = 0.0575677 (* 1 = 0.0575677 loss)
12:01.396266      Train net output #1: loss_refine = 0.00299502 (* 1 = 0.00299502 loss)
12:01.396270      Train net output #2: loss_refine1 = 0.00881063 (* 1 = 0.00881063 loss)
12:01.396273      Train net output #3: loss_refine2 = 0.0100217 (* 1 = 0.0100217 loss)
12:01.396672  Iteration 4340, lr = 0.0001
12:20.180405 32388 solver.cpp:219] Iteration 4360 (1.06471 iter/s, 18.7844s/20 iters), loss = 0.181949
12:20.180433      Train net output #0: loss_midn = 0.142086 (* 1 = 0.142086 loss)
12:20.180439      Train net output #1: loss_refine = 0.00437473 (* 1 = 0.00437473 loss)
12:20.180444      Train net output #2: loss_refine1 = 0.0119189 (* 1 = 0.0119189 loss)
12:20.180447      Train net output #3: loss_refine2 = 0.0133803 (* 1 = 0.0133803 loss)
12:20.532896  Iteration 4360, lr = 0.0001
12:41.684222 32388 solver.cpp:219] Iteration 4380 (0.930057 iter/s, 21.5041s/20 iters), loss = 0.215015
12:41.684250      Train net output #0: loss_midn = 0.00031576 (* 1 = 0.00031576 loss)
12:41.684257      Train net output #1: loss_refine = 0.00440917 (* 1 = 0.00440917 loss)
12:41.684259      Train net output #2: loss_refine1 = 0.00945 (* 1 = 0.00945 loss)
12:41.684263      Train net output #3: loss_refine2 = 0.00943446 (* 1 = 0.00943446 loss)
12:41.684662  Iteration 4380, lr = 0.0001
13:00.646263 32388 solver.cpp:219] Iteration 4400 (1.05473 iter/s, 18.9622s/20 iters), loss = 0.236493
13:00.646291      Train net output #0: loss_midn = 0.152448 (* 1 = 0.152448 loss)
13:00.646297      Train net output #1: loss_refine = 4.01682e-14 (* 1 = 4.01682e-14 loss)
13:00.646301      Train net output #2: loss_refine1 = 0.00930832 (* 1 = 0.00930832 loss)
13:00.646306      Train net output #3: loss_refine2 = 0.0112133 (* 1 = 0.0112133 loss)
13:00.646688  Iteration 4400, lr = 0.0001
13:19.318856 32388 solver.cpp:219] Iteration 4420 (1.07108 iter/s, 18.6728s/20 iters), loss = 0.200657
13:19.318886      Train net output #0: loss_midn = 0.00179278 (* 1 = 0.00179278 loss)
13:19.318891      Train net output #1: loss_refine = 0.00438066 (* 1 = 0.00438066 loss)
13:19.318894      Train net output #2: loss_refine1 = 0.0138651 (* 1 = 0.0138651 loss)
13:19.318898      Train net output #3: loss_refine2 = 0.0102132 (* 1 = 0.0102132 loss)
13:19.532181  Iteration 4420, lr = 0.0001

**Comment by ppengtang** [comment]
Hi, thanks for your question.

Recently, I found that directly setting the maximum image sizes to one of the five scales randomly can produce very promising performance (not worse than the results reported in our paper). To save GPU memory, you can try this small trick by revising [these codes](https://github.com/ppengtang/oicr/blob/pcl/lib/utils/blob.py#L38-L43).

I feel so sorry that I don't remember the losses for different iterations. And I wrongly deleted my log files. But these losses seem reasonable. You can check whether the models are trained correctly by testing the detection performance.

Hope these can help you!
